src/common/utils/setup-helpers.js

```diff
@@ -328,18 +328,48 @@ function checkNativeTextTracksSupport(options: KPOptionsObject): void {
   }
 }
 
+/**
+ * Sets config option for Ads with MSE
+ * @private
+ * @param {KPOptionsObject} options - kaltura player options
+ * @returns {void}
+ */
+function _configureAdsWithMSE(options: KPOptionsObject): void {
+  const preferNativeHls = !!Utils.Object.getPropertyPath(options, 'playback.preferNative.hls');
+  const preferNativeDash = !!Utils.Object.getPropertyPath(options, 'playback.preferNative.dash');
+  if ((!preferNativeHls || !preferNativeDash) && options.plugins && options.plugins.ima) {
+    const playAdsWithMSE = Utils.Object.getPropertyPath(options, 'playback.playAdsWithMSE');
+    const reloadSourceOnComplete = Utils.Object.getPropertyPath(options, 'plugins.ima.reloadSourceOnComplete');
+    const disableMediaPreload = Utils.Object.getPropertyPath(options, 'plugins.ima.disableMediaPreload');
+
+    if (typeof playAdsWithMSE !== 'boolean') {
+      options = Utils.Object.createPropertyPath(options, 'playback.playAdsWithMSE', true);
+    }
+    if (typeof reloadSourceOnComplete !== 'boolean') {
+      options = Utils.Object.createPropertyPath(options, 'plugins.ima.reloadSourceOnComplete', false);
+    }
+    if (typeof disableMediaPreload !== 'boolean') {
+      options = Utils.Object.createPropertyPath(options, 'plugins.ima.disableMediaPreload', true);
+    }
+  }
+}
 /**
  * Sets config option for LG TV
  * @private
  * @param {KPOptionsObject} options - kaltura player options
  * @returns {void}
  */
 function configureLGTVDefaultOptions(options: KPOptionsObject): void {
-  if (isLGTV()) {
+  if (isSmartTv()) {
     const preferNativeHls = Utils.Object.getPropertyPath(options, 'playback.preferNative.hls');
+    const preferNativeDash = Utils.Object.getPropertyPath(options, 'playback.preferNative.dash');
     if (typeof preferNativeHls !== 'boolean') {
-      options = Utils.Object.createPropertyPath(options, 'playback.preferNative.hls', true);
+      options = Utils.Object.createPropertyPath(options, 'playback.preferNative.hls', false);
+    }
+    if (typeof preferNativeDash !== 'boolean') {
+      options = Utils.Object.createPropertyPath(options, 'playback.preferNative.dash', false);
```

Comment:
ditto

Reply:
Right, but If someone will handle the opposite after that it'll not work..
What do you think?

Reply:
if someone will change the player default he will break the player, we have a lot of assumptions on the player defaults